### ISSUE TITLE
BUG: When using the method Model.test_estimators with refit=True it's not using the specified metrics.

### ISSUE THREAD

**Issue by clausvestergaard** [andersbogsnes/ml_tooling#751]
**Describe the bug**
When using the method Model.test_estimators with refit=True, it's not using the specified metrics. Instead it's using the default metric.

Additionally, wouldn't you except it to be refitting using CV as well if specified in the call?

**To Reproduce**
1) Run the code:
```python
from sklearn.dummy import DummyClassifier
from ml_tooling import Model
from ml_tooling.data.load_demo import load_demo_dataset

data = load_demo_dataset("breast_cancer")
data.create_train_test()

estimators = [DummyClassifier()]

clf, results = Model.test_estimators(data, estimators, metrics="roc_auc", refit=True)

assert clf.result.metrics.name == "accuracy"
clf.result
```
2) Verify that the metric is not roc_auc but accuracy.

**Expected behavior**
I'd expect it to keep evaluating with the previous chosen metrics and I'd expect it to keep using CV if chosen previously.

- [X] I have updated ML Tooling to the latest version
- [X] I have checked if this issue has been reported previously

**Comment by andersbogsnes** [comment]
The metric not carrying through to the `best_estimator` is definitely a bug, good catch!

Not sure it makes sense to refit with CV though, for example, the sklearn.GridSearchCV refit parameter does not do a CV refit: https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.GridSearchCV.html

The general idea would be to return one model, which you could do another round of tuning on if you want - though I'm open to arguments for the pros of refitting with CV

**Comment by clausvestergaard** [comment]
Thanks 

Regarding the cross validation - wouldn't refitting on the entire train set with scoring on the test set prevent further tuning? At least in principle :smile: 

I've been using the CV-option as an alternative to splitting the data into an additional validation partition.

And I'd argue that `model.test_estimators` without CV should be used only if you're not doing any tuning afterwards. On the other hand, if you're testing the estimators with CV, you're implicitly saying that you're planning on resuming the exploration of your model(s). And in that case you wouldn't wanna evaluate on the test-data. Hence, refitting should include CV if specified in the first place.

**Comment by clausvestergaard** [comment]
But I gotta say, my confidence in my argument is a bit rattled knowing that scikit-learn doesn't refit using cv :grin:

**Comment by andersbogsnes** [comment]
Having thought about it for a while, I think your point about not touching the test data until later in the process makes sense. 

Gridsearch is for hyper parameter tuning, not model selection and thus the refit at the end makes sense in that you've found the best parameters for the given model. If one wanted to do a algorithm selection on top of that I would expect a CV as well for that.

Given that we have good support for CV results, I think it's fine to allow the user to choose either.

If no one has any further objections, I think we can proceed with the PR

**Comment by clausvestergaard** [comment]
Perfect
I'll update the PR
